usecases/mu/mpc/usecase.go

```diff
@@ -0,0 +1,560 @@
+				continue
+			}
+
+			e.acVoltage[id] = measurements.AddDescription(model.MeasurementDescriptionDataType{
+				MeasurementType: util.Ptr(model.MeasurementTypeTypeVoltage),
+				CommodityType:   util.Ptr(model.CommodityTypeTypeElectricity),
+				Unit:            util.Ptr(model.UnitOfMeasurementTypeV),
+				ScopeType:       util.Ptr(model.ScopeTypeTypeACVoltage),
+			})
+
+			// if constraints are configured for acVoltage[id], set the
+			// constraint id and update measurementsConstraintData
+			if acVoltageConstraints[id] != nil {
+				acVoltageConstraints[id].MeasurementId = e.acVoltage[id]
+				*measurementsConstraintData = append(*measurementsConstraintData, *acVoltageConstraints[id])
+			}
+
+			parameterDescription := model.ElectricalConnectionParameterDescriptionDataType{
+				ElectricalConnectionId:  electricalConnectionId,
+				MeasurementId:           e.acVoltage[id],
+				VoltageType:             util.Ptr(model.ElectricalConnectionVoltageTypeTypeAc),
+				AcMeasuredPhases:        acVoltagePhasesFrom[id],
+				AcMeasuredInReferenceTo: acVoltagePhasesTo[id],
+				AcMeasurementType:       util.Ptr(model.ElectricalConnectionAcMeasurementTypeTypeApparent),
+				AcMeasurementVariant:    util.Ptr(model.ElectricalConnectionMeasurandVariantTypeRms),
+			}
+
+			parameterDescriptionId := electricalConnection.AddParameterDescription(parameterDescription)
+			if parameterDescriptionId == nil {
+				return errors.New("could not add parameter description")
+			}
+		}
+	}
+
+	return nil
+}
+
+func (e *MPC) configureMonitorFrequency(
+	measurements *server.Measurement,
+	electricalConnection *server.ElectricalConnection,
+	electricalConnectionId *model.ElectricalConnectionIdType,
+	measurementsConstraintData *[]model.MeasurementConstraintsDataType,
+) error {
+	if e.frequencyConfig == nil {
+		return nil
+	}
+
+	e.acFrequency = measurements.AddDescription(model.MeasurementDescriptionDataType{
+		MeasurementType: util.Ptr(model.MeasurementTypeTypeFrequency),
+		CommodityType:   util.Ptr(model.CommodityTypeTypeElectricity),
+		Unit:            util.Ptr(model.UnitOfMeasurementTypeHz),
+		ScopeType:       util.Ptr(model.ScopeTypeTypeACFrequency),
+	})
+
+	// if constraints are configured for acFrequency, set the
+	// constraint id and update measurementsConstraintData
+	if e.frequencyConfig.ValueConstraints != nil {
+		e.frequencyConfig.ValueConstraints.MeasurementId = e.acFrequency
+		*measurementsConstraintData = append(*measurementsConstraintData, *e.frequencyConfig.ValueConstraints)
+	}
+
+	parameterDescription := model.ElectricalConnectionParameterDescriptionDataType{
+		ElectricalConnectionId: electricalConnectionId,
+		MeasurementId:          e.acFrequency,
+		VoltageType:            util.Ptr(model.ElectricalConnectionVoltageTypeTypeAc),
+	}
+
+	parameterDescriptionId := electricalConnection.AddParameterDescription(parameterDescription)
+	if parameterDescriptionId == nil {
+		return errors.New("could not add parameter description")
```

Comment:
These error cases could be tested by passing mocks of `server.ElectricalConnection` and return an error on the function call.

Applies to all the other private functions as well.